tests/experimental/metric_repository/test_metric_list_metric_retriever.py

```diff
@@ -0,0 +1,787 @@
+from typing import Dict, List
+
+import pytest
+
+from great_expectations.data_context import CloudDataContext
+from great_expectations.datasource.fluent import BatchRequest
+from great_expectations.datasource.fluent.interfaces import Batch
+from great_expectations.experimental.metric_repository.metric_list_metric_retriever import (
+    MetricListMetricRetriever,
+)
+from great_expectations.experimental.metric_repository.metrics import (
+    ColumnMetric,
+    MetricException,
+    MetricTypes,
+    TableMetric,
+)
+from great_expectations.rule_based_profiler.domain_builder import ColumnDomainBuilder
+from great_expectations.validator.exception_info import ExceptionInfo
+from great_expectations.validator.validator import Validator
+
+pytestmark = pytest.mark.unit
+
+from pytest_mock import MockerFixture
+
+
+def test_get_metrics_table_metrics_only(mocker: MockerFixture):
+    mock_context = mocker.Mock(spec=CloudDataContext)
+    mock_validator = mocker.Mock(spec=Validator)
+    mock_context.get_validator.return_value = mock_validator
+    computed_metrics = {
+        ("table.row_count", (), ()): 2,
+        ("table.columns", (), ()): ["col1", "col2"],
+        ("table.column_types", (), "include_nested=True"): [
+            {"name": "col1", "type": "float"},
+            {"name": "col2", "type": "float"},
+        ],
+    }
+    table_metrics_list = [
+        MetricTypes.TABLE_ROW_COUNT,
+        MetricTypes.TABLE_COLUMNS,
+        MetricTypes.TABLE_COLUMN_TYPES,
+    ]
+    aborted_metrics: Dict[str, str] = {}
+    mock_validator.compute_metrics.return_value = (
+        computed_metrics,
+        aborted_metrics,
+    )
+    mock_batch = mocker.Mock(spec=Batch)
+    mock_batch.id = "batch_id"
+    mock_validator.active_batch = mock_batch
+
+    metric_retriever = MetricListMetricRetriever(context=mock_context)
+
+    mock_batch_request = mocker.Mock(spec=BatchRequest)
+
+    metrics = metric_retriever.get_metrics(
+        batch_request=mock_batch_request,
+        metric_list=table_metrics_list,
+    )
+    assert metrics == [
+        TableMetric[int](
+            batch_id="batch_id",
+            metric_name="table.row_count",
+            value=2,
+            exception=None,
+        ),
+        TableMetric[List[str]](
+            batch_id="batch_id",
+            metric_name="table.columns",
+            value=["col1", "col2"],
+            exception=None,
+        ),
+        TableMetric[List[str]](
+            batch_id="batch_id",
+            metric_name="table.column_types",
+            value=[
+                {"name": "col1", "type": "float"},
+                {"name": "col2", "type": "float"},
+            ],
+            exception=None,
+        ),
+    ]
+
+
+def test_get_metrics_full_list(mocker: MockerFixture):
+    mock_context = mocker.Mock(spec=CloudDataContext)
+    mock_validator = mocker.Mock(spec=Validator)
```

Comment:
Only changes were in this file. We are now using the `mocker` fixture after #9586